core/container/externalbuilder/session.go

```diff
@@ -70,6 +72,7 @@ func (s *Session) waitForExit(logger *flogging.FabricLogger, stderr io.Reader) {
 	defer s.mutex.Unlock()
 	s.exitErr = err
 	s.waitStatus = s.command.ProcessState.Sys().(syscall.WaitStatus)
+	os.RemoveAll(s.launchDir)
```

Comment:
This was the simplest way to keep complexity down and ensure that the resources were cleaned up before returning. Otherwise, I haven't found a simple way to contain the os.RemoveAll calls and guarantee that they've executed before the peer exits.